### DISCUSSION TITLE
About room delete and query

### DISCUSSION THREAD

**Discussion by charlescui** [colyseus/colyseus#796] [Q&A]
How to use the room id to actively delete a room So that matchMaker.query can't find the room data.

I want to get a room instance not a room cache. When I get a room instance, may be I could use room.disconnect() to remove this room instance.

But, I can not get a room instance. I use matchMaker.query, get room cache......

Help me. Thanks!

**Comment by endel** [comment]
Hi @charlescui, it is not recommended to access room instances directly, as when you scale to multiple processes/VMs, the room instance itself is not going to be available.

You can, however, perform a "remote call" in them, via [`matchMaker.remoteRoomCall()`](https://docs.colyseus.io/server/matchmaker/#remoteroomcallroomid-method-args):

```typescript
const rooms = matchMaker.query({/* ... */});

rooms.forEach((room) => {
  matchMaker.remoteRoomCall(room.roomId, "disconnect");
});
```
